### PR TITLE
feat(ui-home): Initial push for home page

### PR DESCRIPTION
**Pull request by amritkv17**
Initial commits for the Home page , Project page, Add Project page: 

1. Added home page with endpoints integrated as below : 

       a.  fetching my projects,
       b.  my subscriptions,
       c.  recent components,
       d.  recent releases.
      
2. Redirection to the project page and from project page to add project page

### CODE REVIEW

**Review by heliocastro — CHANGES_REQUESTED** [review]
Why commitlint was removed ???

**Comment by ag4ums** [comment]
@amritkv17, Please add copyright and license headers in all the new/modified files.

**Comment by heliocastro** [comment]
@amritkv17 Do the copyrights in this way:

```Copyright (C) 2023 SW360 Project Authors (see <https://github.com/eclipse-sw360/sw360/NOTICE>)```

And inside NOTICE file add your company data / personal data

**Review by heliocastro — CHANGES_REQUESTED** [review]
@amritkv17 
Again:
1 - Why commit lint was removed ?
2 - Why now prettier comfig was removed
3 - Why we have a yarn.lock, there's never an agreement about to start use Yarn. ONLY npm should be used
4 - No PNG files, only svg.

**Review by heliocastro — CHANGES_REQUESTED** [review]
Please address all entries

File: src/components/SearchVendorsModal/index.tsx
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/components/SearchVendorsModal/index.tsx:1]: Resolved this issue. :)

File: src/components/footer.tsx [hunk 1]
```diff
@@ -1,23 +1,28 @@
-/*
- * Copyright (C) TOSHIBA CORPORATION, 2023.
- * Copyright (C) Toshiba Software Development (Vietnam) Co., Ltd., 2023.
- * SPDX-License-Identifier: EPL-2.0
- * License-Filename: LICENSE
- */
```
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/components/footer.tsx:6]: I am really sorry for this mistake. Resolved this issue. :)

File: src/utils/table-builder/buildTables.tsx
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/utils/table-builder/buildTables.tsx:1]: Resolved this issue. :)

File: .eslintrc.json
[the remark this answers is not in the document]
Comment (amritkv17) [reply .eslintrc.json:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: .prettierrc.js
[the remark this answers is not in the document]
Comment (amritkv17) [reply .prettierrc.js:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: commitlint.config.js
[the remark this answers is not in the document]
Comment (amritkv17) [reply commitlint.config.js:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: src/utils/pagination/Pagination.tsx
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/utils/pagination/Pagination.tsx:1]: Resolved this issue. :)

File: src/utils/table-builder/buildTable.module.css
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/utils/table-builder/buildTable.module.css:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: src/utils/env.ts
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/utils/env.ts:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: src/utils/pagination/pagination.module.css
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/utils/pagination/pagination.module.css:1]: Thanks for pointing out, it was by mistake. Resolved this issue. :)

File: src/pages/api/auth/[...nextauth].ts
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/pages/api/auth/[...nextauth].ts:1]: Actually, few endpoints are yet to be developed. And due to this, we haven't yet integrated the authentication code written by Toshiba colleagues into our code. That's why it shows removed.

File: src/styles/auth.css
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/styles/auth.css:1]: Actually, few endpoints are yet to be developed. And due to this, we haven't yet integrated the authentication code written by Toshiba colleagues into our code. That's why it shows removed.

File: public/images/logo-edited.svg
[the remark this answers is not in the document]
Comment (amritkv17) [reply public/images/logo-edited.svg:1]: I am working on it. :)

File: public/images/profile.png
[the remark this answers is not in the document]
Comment (amritkv17) [reply public/images/profile.png:1]: I am working on it. :)

File: src/components/SelectCountry.tsx
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/components/SelectCountry.tsx:1]: Thanks for the suggestion. I am working on it. :)

**Comment by amritkv17** [comment]
> @amritkv17 Do the copyrights in this way:
> 
> `Copyright (C) 2023 SW360 Project Authors (see <https://github.com/eclipse-sw360/sw360/NOTICE>)`
> 
> And inside NOTICE file add your company data / personal data

Resolved this issue. :)

**Comment by amritkv17** [comment]
> @amritkv17, Please add copyright and license headers in all the new/modified files.

Thanks for pointing out, it was by mistake. Resolved this issue. :)

**Comment by amritkv17** [comment]
> Why commitlint was removed ???

Thanks for pointing out, it was by mistake. Resolved this issue. :)

**Comment by amritkv17** [comment]
> @amritkv17 Again: 1 - Why commit lint was removed ? 2 - Why now prettier comfig was removed 3 - Why we have a yarn.lock, there's never an agreement about to start use Yarn. ONLY npm should be used 4 - No PNG files, only svg.

Point 1 & 2:  Thanks for pointing out, it was by mistake. Resolved this issue. :)
Point 3 & 4:  I am working on it. :)

**Review by heliocastro — CHANGES_REQUESTED** [review]
Take a look on this:
image

So, PLEASE add back the Toshiba next-auth work, and add a flag option if you do not want automatic auth in the test environment, but NEVER remove work done.

Toshiba made an excellent job using the right approach from the beginning, let's not screw this up, ok ?

File: package.json
[the remark this answers is not in the document]
Comment (heliocastro) [reply package.json:1]: 4 - Did you noticed that you removed coer packages like country-flag icons and next-auth ?

File: src/components/SelectCountry.tsx
Comment (heliocastro) [inline src/components/SelectCountry.tsx:1]: Don't do that **again**
Use something stablished:
https://www.npmjs.com/package/country-list

File: src/pages/_document.tsx [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
   return (
     <Html lang="en">
-      <Head />
+      <Head>
+      <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css"
+              rel="stylesheet" integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD"
+              crossOrigin="anonymous" />
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.5.0/font/bootstrap-icons.css"></link>
+      </Head>
+
```
Comment (heliocastro) [inline src/pages/_document.tsx:20]: Don't do this way.
Use this or similar class:
https://www.npmjs.com/package/react-bootstrap-icons

File: src/services/home.service.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+    const [taskSubmissionData, setTaskSubmissionData] = useState(null);
+    const getHomePageTaskSubmissionData = async () => {
+
+        let myTaskSubmissionFetchURL: string = NEXT_PUBLIC_SW360_API_BASE_URL + '/myTaskSubmissions';
+        const opts: RequestContent = { method: 'GET', headers: {}, body: null };
+
+        opts.headers['Content-Type'] = 'application/json';
+        opts.headers['Authorization'] = `Token IVfIpfWZGuAaDahVNle2`;
+        opts.headers['Access-Control-Allow-Origin'] = '*';
```
Comment (heliocastro) [inline src/services/home.service.ts:151]: **ALERT** Token in code !!!

**Review by heliocastro — CHANGES_REQUESTED** [review]
Why do you removed the empty line on end of all files ?
The large majority of linters complain about files without extra empty line.
Maybe you should config your editor to change this behaviour.

File: src/services/home.service.ts (hunk 3 above)
Comment (amritkv17) [reply src/services/home.service.ts]: Addressed this issue ! :)

File: src/components/SearchVendorsModal/index.tsx
[the remark this answers is not in the document]
Comment (amritkv17) [reply src/components/SearchVendorsModal/index.tsx:1]: Addressed this issue ! :)

**Review by heliocastro — DISMISSED** [review]
Please rebase and then good to go

**Review by heliocastro — APPROVED** [review]
Let's start the grid.js move :-)

**Comment by heliocastro** [comment]
@amritkv17 Please check the validation of commits.
PR is good, no need to change the code, just fix the header

**Comment by amritkv17** [comment]
> @amritkv17 Please check the validation of commits. PR is good, no need to change the code, just fix the header

@heliocastro Thanks ! I am on it. :)

And once this PR is ready, I will incorporate `gridjs` too.

**Comment by heliocastro** [comment]
@amritkv17 You can run this locally to check before commit from inside the top source dir:
```bash
docker run -it --rm -v $(pwd):/github/workspace ghcr.io/apache/skywalking-eyes/license-eye header
```

**Comment by amritkv17** [comment]
> @amritkv17 Please check the validation of commits. PR is good, no need to change the code, just fix the header

@heliocastro Is it necessary to put licensing header in the `.json` files also ?

**Comment by heliocastro** [comment]
If the json file accepts, yes.
If the parser complains, add an exception on the .license file.

But exception is exception, always add a license in any file

**Review by heliocastro — APPROVED** [review]
Finally good to go

**Comment by heliocastro** [comment]
@amritkv17 Is there any commit missing signing ?

**Review by KoukiHama — APPROVED** [review]

**Comment by ag4ums** [comment]
@amritkv17, seems commit history is messed up, please created a new branch from the main and merge the code there.
and close this PR once you create the new one.

**Comment by amritkv17** [comment]
Closing this PR. New Pr is here(https://github.com/eclipse-sw360/sw360-frontend/pull/29)

**Pull request closed without merging**